hal/src/boron/network/sara_ncp_client.cpp

```diff
@@ -268,10 +268,16 @@ int SaraNcpClient::initParser(Stream* stream) {
         // Cellular Global Identity (partial)
         if (r >= 3) {
             auto rat = r >= 4 ? static_cast<CellularAccessTechnology>(val[3]) : self->act_;
+            if (self->ncpId() == PLATFORM_NCP_SARA_R410) {
+                if (rat == CellularAccessTechnology::LTE) {
```

Comment:
This is only use locally and does not matter whether it's LTE, LTE Cat M1 or any other LTE type. We can probably omit the fixup here specifically.

Reply:
For now yes. But this will be used if in the future, we use a SARA with a Cat-1 modem for example, and then differentiate RATs for whatever we plan to do here? 🤔

Reply:
The main thing we are checking here is still whether we are on LTE, no matter its type. So, I don't see why we need to add a fixup here. 7 is still LTE and this is just a local variable.

Reply:
For example a case where, we add additional stuff into `case CellularAccessTechnology::LTE:`, then we want to differentiate the cat-m1 modem and the cat-1/lteonly modem.

Reply:
I doubt we'd ever need to do that here specifically. This value is only used to know whether we are on LTE locally within this function (any type of it) to generate the CGI.

Reply:
If we don't ever do that then that's good. The point I am mentioning here is that if we do something different for LTE vs LTE-Catm1 for a different radio in the future for sara, then the new code will look like this, and we might need a differentiation between R410 and other LTE radios in that case...........? 
```
case CellularAccessTechnology::LTE:
       // CGI data + something specific for an LTE modem only
case CellularAccessTechnology::LTE_CAT_M1:
      // CGI data
case CellularAccessTechnology::LTE_NB_IOT: 
       // CGI data
       break;
```

Reply:
I understand that possibility, however for all intents and purposes the data here specifically returned by `AT+CEREG` is exactly the same no matter which LTE type we are on.

If you are really keen on fixing this everywhere, let's make it a bit more cleaner and introduce some kind of a function which would take RAT number and modem type and would correctly return `CellularAccessTechnology` for us.